088-watching-multiple-namespaces.md

```diff
@@ -0,0 +1,155 @@
+First proposal on `KafkaTopic` resource: [issue #1206](https://github.com/strimzi/strimzi-kafka-operator/issues/1206)
+For me, the goal of a kubernetes operator it's the capacity to create and configure multiples ressources accross kubernetes namespaces.
+
+## Current situation
+
+For now, Strimzi has one capacity. In facts, for `KafkaUser` and `KafkaTopic` Strimzi deploy a `user-operator` and `topic-operator` in the namespace where the `Kafka` resource resides.
+For `KafkaConnector` it's almost the same thing except that `cluster-operator` has the capacity to detect `kafkaConnector` resources in all namespaces, but after have found a `KafkaConnector` resource it search a `KafkaConnect` cluster in the namespace of this ressources and it doesn't exists because the `KafkaConnector` is deploy in the same namespace of `Kafka` resource.
+
+## Motivation
+
+In a multi-tenant Kubernetes cluster, it'as a common practice to define authorization at namespace level or to deploy many application in different namespace. In the case where `Kafka` (Generaly it's case) it's mutualized for many applications and/or for many customers, it's more flexible and secure for deployed each applications in a dedicated namespace.
+
+With Strimzi we have the capacity to create a Helm chart for our applications which automatically deploye and configure the customer KafkaUser, kafkaTopic and KafkaConnector, but the problem is than all customer must be deployed in the same namespace where `Kafka` resource is deployed and so in the same namespace of all customer. So with thousands of customers the read and management of this namespace become difficult or impossible to use while having thousands namespaces it's easier to manage. Of Kubernetes point of view this allows to separate `Kafka` components in dedicated namespace, configure quota and limit on namespace, secure namespace (customers and Kafka namespace) with network policies but particularly to secure and separate all client applications that use Kafka.
+
+
+## Proposal
+
+The next proposal retake this first proposal [issue #1206](https://github.com/strimzi/strimzi-kafka-operator/issues/1206) but attempt to apply it on all resources (`kafkaUser, KafkaTopic and KafkaConnector`).
+
+
+### KafkaTopic
+
+Create a new `CustomResourceDefinition` to handle topics that are created in namespaces, called `KafkaNamespaceTopic`. This CRD will allow the user to create a topic that will be automatically namespaced within Kafka by the Topic Operator. It will encode the actual topic name using the Kubernetes namespace the resource resides, and either the `metadata.name` or `spec.topicName` (if defined) of the instance of a `KafkaNamespaceTopic`.
+
+`{Kubernetes namespace name}.{metadata.name|spec.topicName}`
```

Comment:
One more thought ... While for users, prefixing or suffixing makes sense, for Topics, it is way more controversial. Because it means that once you have the topic created in namespace A and have some data in it, you cannot move it to another namespace (if you just create the same topic in another namespace it would get a new name and be empty). Moreover, the Topic Operator already uses the `.spec.topicName` field and is able to handle conflicts. So the question is whether the prefixing or suffixing here makes really sense and is desired.